### DISCUSSION TITLE
Disable cgroups v1

### DISCUSSION THREAD

**Discussion by chancez** [actions/virtual-environments#5941] [General questions]
Is there a way to disable cgroups v1, either using an alternative image or some other means? Trying to run certain eBPF tools within Kubernetes via KIND (kubernetes-in-docker) doesn't completely work when cgroups v1 is enabled, as it causes some issues.

**Comment by mikhailkoliada** [comment]
@chancez Hello!

cgroups is a kernel-provided mechanism and therefore it can only be managed via the `/proc/cmdline` feature (aka kernel boot param which reside in the grub bootloader in our case and can't be applied without physical agent reboot), that said, cgroups can not be disabled on the hosted runners.

**Reply by chancez** [reply]
To be clear, I'm not asking to disable it entirely, just v1. I still want/need cgroups v2, but only v2 without v1. Is there any plans to or ways this might be possible in the future?

**Reply by mikhailkoliada** [reply]
There were no strict requests to disable v1 in the past, I think it is possible for 20.04 and 22.04, as of 18 I am not sure as the software compatibility with v2 needs to be checked more precisely.

**Reply by chancez** [reply]
A variant with cgroups v2 only with Ubuntu 20.04 or 22.04 is more than enough for my purposes at least, even just 22.04 would be fine. It's mostly to test some eBPF related functionality with KIND. 

At the moment we resort to using a Mac GitHub runner and run our own VM (via Vagrant) within that runner to work-around this 😭 .

**Reply by mikhailkoliada** [reply]
@chancez we will investigate possibility to disable cgroupsv1 in ubuntu 22.04

**Comment by vpolikarpov-akvelon** [comment]
Hi, @chancez. Sorry for such a long delay. Could you please explain what exact problem did you have with cgroups v1? As I understand Ubuntu 22 doesn't mount any controllers on a cgroups v1 hierarchy by default. Doesn't it mean that cgroups v1 are disabled?

**Reply by chancez** [reply]
To be honest, it's slightly over my head. I work at Isovalent and we use KIND to run Kubernetes inside docker for our CI jobs. We build a Kubernetes CNI that requires groups v2 and there's a compatibility issue when certain cgroup v1 controllers are enabled.

See the note just under the `helm install` commands in our docs for rationale: https://docs.cilium.io/en/v1.13/installation/kind/#install-cilium. TL;DR there is a [bug in the kernel < 5.14](https://github.com/torvalds/linux/commit/8520e224f547cd070c7c8f97b1fc6d58cff7ccaa) where cgroup v2 has issues when cgroupv1 net_cls/net_prio are enabled.

**Reply by vpolikarpov-akvelon** [reply]
Thank you for the explanation. I've checked Ubuntu 22.04 runners configuration. Looks like they run 5.15 kernel from the very beginning so it shouldn't be an issue there. Also I can confirm that unified cgroup hierarchy is enabled and docker daemon is configured with 'private' mode for cgroupns as required in cilium docs (actually, it's default and we don't change it).

We can try disabling net_cls/net_prio cgroup v1, but we need some way to check whether it helps or not. Could you provide any information on how the issues you mentioned can be reproduced?

**Reply by chancez** [reply]
I just synced up with our team and they said it should be working as expected at this point then.

**Reply by vpolikarpov-akvelon** [reply]
Hey @chancez. Do I understand you right that everything is working on Ubuntu 22 runners as expected now?

**Reply by chancez** [reply]
I believe so.

**Reply by vpolikarpov-akvelon** [reply]
I'm glad to hear that. I'm closing this thread then. Feel free to reach out if you need more help.
